sirepo/http_request.py

```diff
@@ -91,7 +91,7 @@ def parse_post(**kwargs):
     if kwargs.pkdel('fixup_old_data'):
         r = simulation_db.fixup_old_data(r)[0]
     res.pkupdate(req_data=r)
-    kwargs.pksetdefault(type=True)
+    kwargs.pksetdefault(type=True, require_sim_type=True)
```

Comment:
I think I need this explained to me.

Reply:
Certain api's need to be accessed for a sim type before the role for that sim type has been granted. For example, if a user visits `/jupyter` and they don't have the role they will be redirected to `/jupyterhublogin#/moderation-info`. In order for that route to be loaded we need to call [`api_simulationSchema`](https://github.com/radiasoft/sirepo/blob/cf35e963cd3086be50f00f32e2e9b77a004df3ce/sirepo/server.py#L518). But, if require_sim_type was True then that api request would fail because they don't have the role for jupyterhublogin. So we'd be in a loop (can't get to jupyter because they don't have the role and can't get to the page to request the role because they don't have the role).

In addition, [admins shouldn't be required](https://github.com/radiasoft/sirepo/blob/cf35e963cd3086be50f00f32e2e9b77a004df3ce/sirepo/auth_role_moderation.py#L86) to have the role for the sim_type they are granting a user to use.

Reply:
This is a slippery slope. These type of operations should not depend on accessing the sim_type. That's an ngui problem. Too much tied up in app vs sim_type.